example/accel/trackingmanager-offload.cc

```diff
@@ -0,0 +1,209 @@
+#include <G4EmStandardPhysics.hh>
+#include <G4Gamma.hh>
+#include <G4LogicalVolume.hh>
+#include <G4Material.hh>
+#include <G4PVPlacement.hh>
+#include <G4ParticleDefinition.hh>
+#include <G4ParticleGun.hh>
+#include <G4ParticleTable.hh>
+#include <G4Positron.hh>
+#include <G4Region.hh>
+#include <G4RegionStore.hh>
+#include <G4RunManagerFactory.hh>
+#include <G4SystemOfUnits.hh>
+#include <G4Threading.hh>
+#include <G4ThreeVector.hh>
+#include <G4Track.hh>
+#include <G4TrackStatus.hh>
+#include <G4Types.hh>
+#include <G4UserEventAction.hh>
+#include <G4UserRunAction.hh>
+#include <G4UserTrackingAction.hh>
+#include <G4VUserActionInitialization.hh>
+#include <G4VUserDetectorConstruction.hh>
+#include <G4VUserPrimaryGeneratorAction.hh>
+#include <accel/AlongStepFactory.hh>
+#include <accel/LocalTransporter.hh>
+#include <accel/SetupOptions.hh>
+#include <accel/SharedParams.hh>
+#include <accel/SimpleOffload.hh>
+#include <accel/TrackingManagerOffload.hh>
+#include <corecel/Macros.hh>
+#include <corecel/io/Logger.hh>
+
+namespace
+{
+//---------------------------------------------------------------------------//
+// Global shared setup options
+celeritas::SetupOptions setup_options;
+// Shared data and GPU setup
+celeritas::SharedParams shared_params;
+// Thread-local transporter
+G4ThreadLocal celeritas::LocalTransporter local_transporter;
+
+// Simple interface to running celeritas
+G4ThreadLocal celeritas::SimpleOffload simple_offload;
+
+//---------------------------------------------------------------------------//
+class DetectorConstruction final : public G4VUserDetectorConstruction
+{
+  public:
+    DetectorConstruction()
+        : aluminum_{new G4Material{
+            "Aluminium", 13., 26.98 * g / mole, 2.700 * g / cm3}}
+    {
+        setup_options.make_along_step = celeritas::UniformAlongStepFactory();
+    }
+
+    G4VPhysicalVolume* Construct() final
+    {
+        CELER_LOG_LOCAL(status) << "Setting up detector";
+        auto* box = new G4Box("world", 1000 * cm, 1000 * cm, 1000 * cm);
+        auto* lv = new G4LogicalVolume(box, aluminum_, "world");
+        auto* pv = new G4PVPlacement(
+            0, G4ThreeVector{}, lv, "world", nullptr, false, 0);
+        return pv;
+    }
+
+  private:
+    G4Material* aluminum_;
+};
+
+//---------------------------------------------------------------------------//
+class EMPhysicsConstructor final : public G4EmStandardPhysics
+{
+  public:
+    using G4EmStandardPhysics::G4EmStandardPhysics;
+
+    void ConstructProcess() override
+    {
+        G4EmStandardPhysics::ConstructProcess();
+
+        // Add Celeritas tracking manager to electron, positron, gamma.
+        auto* celer_tracking = new celeritas::TrackingManagerOffload(
+            &shared_params, &local_transporter);
+        G4Electron::Definition()->SetTrackingManager(celer_tracking);
+        G4Positron::Definition()->SetTrackingManager(celer_tracking);
+        G4Gamma::Definition()->SetTrackingManager(celer_tracking);
```

Comment:
Does anyone take ownership of this pointer or is it leaked?

Reply:
The deletion of the pointer is done in the destructor of `G4VUserPhysicsList` via the [`RemoveTrackingManager` member function](https://gitlab.cern.ch/geant4/geant4/-/blob/v11.1.3/source/run/src/G4VUserPhysicsList.cc?ref_type=tags#L326-356). That's analogous to what's done for process managers, albeit making the ownership non-definite.